packages/beacon-node/src/api/impl/beacon/state/utils.ts

```diff
@@ -138,28 +138,32 @@ export function getStateValidatorIndex(
 ): StateValidatorIndexResponse {
   let validatorIndex: ValidatorIndex | undefined;
   if (typeof id === "string") {
+    // mutate `id` and fallthrough to below
     if (id.startsWith("0x")) {
-      // mutate `id` and fallthrough to below
       try {
         id = fromHexString(id);
       } catch (e) {
         return {valid: false, code: 400, reason: "Invalid pubkey hex encoding"};
       }
     } else {
-      validatorIndex = Number(id);
-      // validator is invalid or added later than given stateId
-      if (!Number.isSafeInteger(validatorIndex)) {
-        return {valid: false, code: 400, reason: "Invalid validator index"};
-      }
-      if (validatorIndex >= state.validators.length) {
-        return {valid: false, code: 404, reason: "Validator index from future state"};
-      }
-      return {valid: true, validatorIndex};
+      id = Number(id);
+    }
+  }
+
+  if (typeof id === "number") {
+    validatorIndex = id;
```

Comment:
Having 2 declaration for `validatorIndex` with smaller scope would improve readability and prevent  the issue you fixed.

Reply:
The issue was really there because of a wrong type cast, so by removing that it should not be possible anymore. But your suggestion makes sense, simpler to follow statue of validatorIndex, as it really doesn't need to be declared at the top of the function